### ISSUE TITLE
DTLS: Retransmission handled incorrectly ("Received unexpected record version")

### ISSUE THREAD

**Issue by Thiesius** [randombit/botan#2316]
In the beginning of the handshake the client sends ClientHello message with DTLS Record Version 1.0. The version of the inner message is 1.2. Server sets pending state to DTLS version 1.2. Client meanwhile retransmits the record again, thinking the packet got lost. Retransmitted message arrives to the server but the record version is checked against the pending state and obviously 1.0 != 1.2. As the consequence connection gets alerted while the message probably should be just dropped.

I created some extra info from our test cert environment.

This is the area of code causing the issue. I have added some vars to watch.
![watch](https://user-images.githubusercontent.com/12806671/78234074-ead6f200-74d6-11ea-8a78-060cdfbdad1f.png)

This zipfile contains wireshark dump of the handshake and dumped m_record_buf (which proves that it matches with the ClientHello message in the wireshark)
[dumps.zip](https://github.com/randombit/botan/files/4420161/dumps.zip)

**Comment by randombit** [comment]
Makes sense and yes for DTLS we should just drop it instead of alerting.

**Comment by randombit** [comment]
Won't be fixed in time for 2.14.0 (release next Monday) but will address this in 2.15

**Comment by seppy87** [comment]
I am still receiving this Exception in 2.15 :/

**Comment by seppy87** [comment]
sorry found the error. ok it works

**Comment by volok-aleksej** [comment]
Hi, guys. This bug is fixed [next pull request](https://github.com/randombit/botan/pull/4037).
Please check it.
for reproducing it i have created simple application
[botan_bug.tar.gz](https://github.com/randombit/botan/files/15177052/botan_bug.tar.gz)
